commit 183f732c3f3f307d5673e17b69de6894e1dd2918
Author: Johan Hedberg <johan.hedberg@nokia.com>
Date:   Mon Dec 6 15:56:17 2010 +0200

    Bluetooth: Fix initial RFCOMM DLC security level
    
    Due to commit 63ce0900 connections initiated through TTYs created with
    "rfcomm bind ..." would have security level BT_SECURITY_SDP instead of
    BT_SECURITY_LOW. This would cause instant connection failure between any
    two SSP capable devices due to the L2CAP connect request to RFCOMM being
    sent before authentication has been performed. This patch fixes the
    regression by always initializing the DLC security level to
    BT_SECURITY_LOW.
    
    Signed-off-by: Johan Hedberg <johan.hedberg@nokia.com>
    Acked-by: Luiz Augusto von Dentz <luiz.dentz-von@nokia.com>
    Signed-off-by: Gustavo F. Padovan <padovan@profusion.mobi>

diff --git a/net/bluetooth/rfcomm/core.c b/net/bluetooth/rfcomm/core.c
index fa642aa652bd..432a9a633e8d 100644
--- a/net/bluetooth/rfcomm/core.c
+++ b/net/bluetooth/rfcomm/core.c
@@ -311,6 +311,7 @@ static void rfcomm_dlc_clear_state(struct rfcomm_dlc *d)
 	d->state      = BT_OPEN;
 	d->flags      = 0;
 	d->mscex      = 0;
+	d->sec_level  = BT_SECURITY_LOW;
 	d->mtu        = RFCOMM_DEFAULT_MTU;
 	d->v24_sig    = RFCOMM_V24_RTC | RFCOMM_V24_RTR | RFCOMM_V24_DV;